tracer/src/Datadog.Trace/Util/HexString.cs

```diff
@@ -19,7 +19,10 @@ internal static class HexString
 {
 #if !NETCOREAPP3_1_OR_GREATER
     [ThreadStatic]
-    private static byte[]? _buffer8; // always 8 bytes for ulong
+    private static byte[]? _buffer16; // always 16 bytes for TraceId
+
+    [ThreadStatic]
+    private static byte[]? _buffer8; // always 8 bytes for UInt64 (ulong)
```

Comment:
Maybe have a single buffer that you resize if it's too small?

Reply:
Basically something like:

```csharp

[ThreadStatic]
private static byte[] _buffer;

private static byte[] GetBuffer(int size)
{
    if (_buffer == null || _buffer.Length < size)
    {
        _buffer = new byte[size];
    }
    
    return _buffer;
}
```

Worst case scenario, you have the same number of allocations but you keep fewer arrays around. Best case scenario, you also reduce the number of allocations.

Reply:
Good idea, thanks.

Reply:
Done! I was also needed to pass a byte count argument around, but instead switched `byte[]` to `ArraySegment<byte>`. This code was modified for .NET Framework, and `ArraySegment<byte>` seems like a better analogue for `ReadOnlySpan<byte>`.